x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/advanced_options_section.tsx

```diff
@@ -0,0 +1,303 @@
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSwitch,
+  EuiAccordion,
+  EuiHorizontalRule,
+  EuiText,
+  EuiSpacer,
+  EuiFieldText,
+  EuiFieldNumber,
+  EuiSelect,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import type { OutputFormInputsType } from './use_output_form';
+
+export interface AdvancedOptionsSectionProps {
+  enabled: boolean;
+  inputs: OutputFormInputsType;
+}
+
+export const AdvancedOptionsSection: React.FunctionComponent<AdvancedOptionsSectionProps> = ({
+  enabled,
+  inputs,
+}) => {
+  const {
+    diskQueueEnabledInput,
+    diskQueuePathInput,
+    diskQueueMaxSizeInput,
+    loadBalanceEnabledInput,
+    diskQueueEncryptionEnabled,
+    diskQueueCompressionEnabled,
+    compressionLevelInput,
+    memQueueEvents,
+    queueFlushTimeout,
+    maxBatchBytes,
+  } = inputs;
+
+  return enabled ? (
+    <EuiAccordion
+      id="advancedOutputOptions"
+      arrowDisplay="left"
+      buttonContent={
+        <FormattedMessage
+          id="xpack.fleet.settings.editOutputFlyout.advancedOptionsToggleLabel"
+          defaultMessage="Advanced options"
+        />
+      }
+    >
+      <>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...maxBatchBytes.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.maxBatchSizeDescriptionLabel"
+              defaultMessage="Maximum Batch Size"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...maxBatchBytes.props} placeholder="Batching Bytes" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.maxBatchSizeDescription"
+                  defaultMessage="Data will be sent to the output when the agent has events which total larger then this configured maximum."
+                />
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFormRow>
+        <EuiSpacer size="m" />
+        <EuiFormRow
+          fullWidth
+          {...queueFlushTimeout.formRowProps}
+          label={
+            <FormattedMessage
+              id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutLabel"
+              defaultMessage="Flush Timeout"
+            />
+          }
+        >
+          <EuiFlexGroup alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiFieldNumber {...queueFlushTimeout.props} placeholder="Seconds" min={0} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutDescription"
+                  defaultMessage="Upon expiry the output queue is flushed and data is written to the output."
+                />
```

Comment:
I think calling this a timeout is confusing. Should we instead label it as a frequency? Not sure exactly how it's implemented though.

```suggestion
            <FormattedMessage
              id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutLabel"
              defaultMessage="Flush Frequency"
            />
          }
        >
          <EuiFlexGroup alignItems="flexStart">
            <EuiFlexItem>
              <EuiFieldNumber {...queueFlushTimeout.props} placeholder="Seconds" min={0} />
            </EuiFlexItem>
            <EuiFlexItem>
              <EuiText>
                <FormattedMessage
                  id="xpack.fleet.settings.editOutputFlyout.queueFlushTimeoutDescription"
                  defaultMessage="Frequency at which data in the queue is written to the output."
                />
```